tensorboard/webapp/widgets/data_table/data_table_component.ts

```diff
@@ -302,59 +308,18 @@ export class DataTableComponent implements OnDestroy, AfterContentInit {
 
   onColumnSelectorClosed() {
     this.insertColumnTo = undefined;
-    this.columnSelector.deactivate();
   }
 
   canContextMenuRemoveColumn() {
     return this.contextMenuHeader?.removable;
   }
 
-  canContextMenuInsert() {
-    return (
-      this.selectableColumns &&
-      this.selectableColumns.length &&
-      this.contextMenuHeader?.movable &&
-      this.contextMenuHeader?.type === 'HPARAM'
-    );
-  }
-
-  isContextMenuEmpty() {
-    return (
-      !this.contextMenuHeader?.removable &&
-      !this.contextMenuHeader?.sortable &&
-      !this.canContextMenuInsert() &&
-      !this.contextMenuHeader?.filterable
-    );
-  }
-
-  contextMenuRemoveColumn() {
-    if (this.contextMenuHeader === undefined) {
-      return;
-    }
-    this.removeColumn.emit(this.contextMenuHeader);
+  onRemoveColumn(header: ColumnHeader) {
+    this.removeColumn.emit(header);
     this.contextMenu.close();
     this.filterModal?.close();
   }
 
-  private getInsertIndex() {
-    if (
-      this.contextMenuHeader === undefined ||
-      this.insertColumnTo === undefined
-    ) {
-      return undefined;
-    }
-
-    const index = this.headers.indexOf(this.contextMenuHeader);
-    if (this.insertColumnTo === Side.LEFT) {
-      return index;
-    }
-    if (this.insertColumnTo === Side.RIGHT) {
-      return Math.min(index + 1, this.headers.length);
-    }
-
-    return index;
-  }
-
   onColumnAdded(header: ColumnHeader) {
```

Comment:
I played with your changes. "Insert Column Left" no longer works as expected. It inserts to the right instead.

Reply:
Oh thanks for catching this! To clarify, the quoted "getInsertIndex()" code should be irrelevant as it's unused, but the other changes introduced a race condition where insertColumnTo is reset to undefined after it is first properly set on column open.

Added a temporary fix (don't reset insertColumnTo to undefined - this is actually unnecessary) and included a simple regression test. Note that this class of problem should be irrelevant in the next PR that creates modals dynamically.